### ISSUE TITLE
BigGAN: gradient accumulation

### ISSUE THREAD

**Issue by gwern** [tensorfork/tensorfork#16]
The BigGAN paper shows that batch size is a critical hyperparameter for the final quality of a BigGAN: the largest the batch, the better the FID/IS. Apparently smaller batches risk permanent forgetting / mode dropping of datapoints, and the bigger the better up to _n_=4096 (highest tested). As models get larger, we will be forced to use _n_=1 GANs per TPU for 512px/1024px BigGANs, or minibatches of _n_=512 at most (since we can get TPU-512s but TPU-1024s preempt/error on us). That's 8x smaller than the best BigGAN result's necessary batch size, and would come at noticeable damage to model quality.

So we want to be able to accumulate gradients across at least 8 minibatches to enable the highest-quality final production runs. Currently compare_gan does not support this.
